tests/strings/test_z_algorithm.py

```diff
@@ -25,14 +22,26 @@ def generate_words(char_set, length):
 
 def test_z_algorithm__corner_cases():
     for c in range(1, 4):
-        for length in range(1, 11):
-            for word in generate_words(ascii_lowercase[:c], length):
-                assert brute_force(word) == z_function(word)
+        for n in range(1, 11):
+            for text in generate_words(ascii_lowercase[:c], n):
+                assert z_function(text) == [get_lcp(text, i) for i in range(n)]
 
 
 def test_z_algorithm__large_cases():
```

Comment:
I don't really see the point of having this test. Testing on random data isn't all that useful. The important thing I think should be to do something like testing all binary strings of length <= 10 and all ternary strings of length <= 7.

Reply:
Yup, testing correctness is not the primary purpose. But rather whether the algorithm can process large strings efficiently. Already checked all possible strings of smaller lengths in the other test.

Reply:
Problem is that on random data. Even the basic brute force algorithm will run in O(n). So unfortunately it is not a good test for efficiency. Something like `'ab' * 10 ** 5` will probably do a lot better. At least for that one the brute solution will run in O(n^2).

Reply:
Right about that. That's why there is a test like `a * 10**5`. I can also add the `ab * 10**5` case if you still feel this is necessary? Brute force times out with the current test cases.

Reply:
`'ab' * 10**5` is definitely more of an interesting test case than `'a' * 10**5`. Add it if you want.